src/main/java/httpserver/HTTPRequest.java

```diff
@@ -25,509 +25,502 @@
-    Map<String, String> out = new HashMap<String, String>();
-    for (String item : data) {
-      if (item.indexOf("=") == -1) {
-        out.put(item, null);
-        continue;
-      }
-
-      String value = item.substring(item.indexOf('=') + 1);
-
-      /*  Attempt to URL decode the value, because it *might* be user input.
-          If it can't be decoded, it doesn't matter, the original, undecoded
-          value is still used.
-       */
-      try {
-        value = URLDecoder.decode(value, "UTF-8");
-      }
-      catch (UnsupportedEncodingException e) {}
-
-      out.put(item.substring(0, item.indexOf('=')), value);
-    }
-
-    return out;
-  }
-
-  /**
-   * Figure out what kind of HTTPHandler you want, based on the path. <p>
-   *
-   * This uses the statically set {@link HTTPRouter} to determine the
-   * correct HTTPHandler to be used for the current request. If there isn't
-   * a statically set HTTPRouter, a 500 error is sent back to the
-   * client.
-   *
-   * @return a new instance of some form of HTTPHandler.
-   *
-   * @see HTTPRouter
-   * @see HTTPRouter#determineHandler
-   * @see HTTPHandler
-   */
-  public HTTPHandler determineHandler() throws HTTPException {
-    if (router == null) {
-      return new DeathHandler();
-    }
-
-    if (isType(POST_REQUEST_TYPE) && getPostData().isEmpty()) {
-      return new MessageHandler(411, "You made a POST request without any data...");
-    }
-    
-    String path = getSplitPath().isEmpty() ? "" : getSplitPath().get(0);
-    return router.route(path, this);
-  }
-
-  /**
-   * Return if the request type is the passed in type.
-   * @param requestTypeCheck The type to check.
-   * @return whether the request type equals the passed in String.
-   */
-  public boolean isType(String requestTypeCheck) {
-    return getRequestType().equalsIgnoreCase(requestTypeCheck);
-  }
-
-  /**
-   * Sets the requestLine, and all derived items. <p>
-   * 
-   * Based off of the passed in line, the request type, request path, and
-   * request protocol can be set.
-   *
-   * @param line  The first line in an HTTP request. Should be in
-   *              {@code [type] [full path] [protocol]} form.
-   * @throws HTTPException  When the first line does not contain two spaces,
-   *                        signifying that the passed in line is not in
-   *                        HTTP 1.1. When the type is not an expected type
-   *                        (currently GET, POST, and HEAD).
-   *
-   * @see HTTPRequest#setRequestType
-   * @see HTTPRequest#setFullPath
-   * @see HTTPRequest#setRequestProtocol
-   */
-  public void setRequestLine(String line) throws HTTPException {
-    this.requestLine = line;
-
-    /*  Split apart the request line by spaces, as per the protocol.
-        The request line should be:
-          [request type] [path] [protocol]
+    public HTTPHandler determineHandler() throws HTTPException {
+        if (router == null) {
+            return new DeathHandler();
```

Comment:
I think this only happens if you don't have a router (`if (router == null)`), which would mean they haven't specified any handlers (or they did, but they didn't attach the router to the server).

If this ever happens we might want to throw some sort of `RuntimeException` or something, instead of saying everything's a-ok.